Sources/Datadog/Core/Upload/DataUploadWorker.swift

```diff
@@ -102,22 +94,21 @@ extension DataUploadConditions.Blocker: CustomStringConvertible {
     var description: String {
         switch self {
         case let .battery(level: level, state: state):
-            return "🔋 Battery: \(state) \(level)%"
+            return "🔋 Battery state is: \(state) (\(level)%)"
         case .lowPowerModeOn:
-            return "🔌 Low Power Mode: on"
+            return "🔌 Low Power Mode is: enabled"
         case let .networkReachability(description: description):
-            return "📡 Reachability: " + description
+            return "📡 Network reachability is: " + description
         }
     }
 }
 
-extension DataUploadConditions.Report: CustomStringConvertible {
+extension Array where Element == DataUploadConditions.Blocker {
```

Comment:
nit: those two `extensions` can be `private` (or `fileprivate`) to avoid indexing them in the global namespace. 

On a longer term, keeping narrow namespaces should give faster compilation time if the _whole module optimisation_ is enabled (impossible to spot on individual usage).

Reply:
Thanks for the reminder, I overlooked that aspect! I'll correct that.